src/derived/slot.rs

```diff
@@ -206,14 +206,25 @@ where
 
                 return Ok(value);
             }
-        }
+
+            memo.value.take()
```

Comment:
I see. This change in behavior with respect to panics applies regardless of whether update is used or not. Nonetheless, I think it's ok. I think dropping the cached value is a reasonable recovery strategy for panics.